packages/components/src/ListView/Items/Items.component.js

```diff
@@ -35,17 +56,25 @@ export class ItemsComponent extends React.PureComponent {
 	}
 
 	getRowHeight({ index }) {
+		if (this.props.getItemHeight) {
+			// If an height computation has been provided, use the "old"
+			// way of computing row height with the provided computation
+			return this.oldGetRowHeight.call(this, { index });
+		}
+
 		if (this.hasToggleAll && index === 0) {
-			return 40;
+			return TOGGLE_ALL_ROW_HEIGHT;
 		}
 
-		let extraHeight = 0;
 		const currentItem = this.getItemByIndex(index);
+
 		if (currentItem && currentItem.children && currentItem.expanded) {
-			extraHeight = currentItem.children.length * this.props.getItemHeight();
+			return (currentItem.children.length + 1) * ROW_HEIGHT + ROW_INNER_PADDINGS;
 		}
 
-		return this.props.getItemHeight() + extraHeight;
+		const isLastItem = index === this.props.items.length - Number(!this.hasToggleAll);
```

Comment:
Could you please explain this one a bit more. I get confused with `!` together with `Number()`. You take the total number of items and subtract Number(boolean)? 🤔 Is there another way to express this one?

Reply:
I wanted to use the same logic as used above (in the code). In the case where ther is a "toggle all" item, the last item's index will equal `items.length`, it would be `items.length` - 1 otherwise

Reply:
I see, if it's already used elsewhere that's fine I guess. 

I think it may still be easier to follow if it's seen as two conditions, similar to the if condition just before this one 
```
if (this.hasToggleAll && index === 0)
``` 

So in this case, what do you think of this (seeing as it also has some previous code like it):
```
const isLastItem = index === this.props.items.length && this.hasToggleAll;
```
(I don't manage to convert it on the spot, not sure if it should be `this.hasToggleAll` or `!this.hasToggleAll`, and if it should match the length or be `this.props.items.length - 1`) 😄